PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/serverexplore/ui/SparkServerlessProvisionDialog.java

```diff
@@ -89,73 +94,91 @@ public SparkServerlessProvisionDialog(@NotNull SparkServerlessADLAccountNode adl
+                            @Override
+                            protected void textChanged(DocumentEvent e) {
+                                if (!comp.isLegal()) {
+                                    getOKAction().setEnabled(false);
+                                } else if (isAllFieldsLegal()) {
+                                    getOKAction().setEnabled(true);
+                                }
+                            }
+                        }));
+        getOKAction().setEnabled(false);
         this.getWindow().addWindowListener(new WindowAdapter() {
             @Override
             public void windowOpened(WindowEvent e) {
-                // update totalAU before displaying the dialogue
-                ctrlProvider.updateTotalAU();
+                updateAvailableAU();
+                updateTotalAU();
+                updateCalculatedAU();
                 super.windowOpened(e);
             }
         });
     }
 
+    private void updateAvailableAU() {
+        ctrlProvider.getAvailableAU().subscribe(au -> availableAUField.setText(String.valueOf(au)));
+    }
+
+    private void updateTotalAU() {
+        ctrlProvider.getTotalAU().subscribe(au -> totalAUField.setText(String.valueOf(au)));
+    }
+
+    private void updateCalculatedAU() {
+
+        calculatedAUField.setText(String.valueOf(ctrlProvider.getCalculatedAU(
+                Integer.valueOf(masterCoresField.getText()),
+                Integer.valueOf(workerCoresField.getText()),
+                Integer.valueOf(masterMemoryField.getText()),
+                Integer.valueOf(workerMemoryField.getText()),
+                Integer.valueOf(workerNumberOfContainersField.getText()))));
+    }
+
     // Data -> Components
     @Override
     public void setData(@NotNull SparkServerlessClusterProvisionSettingsModel data) {
-        clusterNameField.setText(data.getClusterName());
-        adlAccountField.setText(data.getAdlAccount());
-        sparkEventsField.setText(data.getSparkEvents());
-        availableAUField.setText(String.valueOf(data.getAvailableAU()));
-        totalAUField.setText(String.valueOf(data.getTotalAU()));
-        calculatedAUField.setText(String.valueOf(data.getCalculatedAU()));
-        masterCoresField.setText(String.valueOf(data.getMasterCores()));
-        masterMemoryField.setText(String.valueOf(data.getMasterMemory()));
-        workerCoresField.setText(String.valueOf(data.getWorkerCores()));
-        workerMemoryField.setText(String.valueOf(data.getWorkerMemory()));
-        workerNumberOfContainersField.setText(String.valueOf(data.getWorkerNumberOfContainers()));
-
-        clusterNameLabel.setText(data.getClusterNameLabelTitle());
-        adlAccountLabel.setText(data.getAdlAccountLabelTitle());
-        SparkEventsLabel.setText(data.getSparkEventsLabelTitle());
-        storageRootPathLabel.setText(data.getStorageRootPathLabelTitle());
-        workerNumberOfContainersLabel.setText(data.getWorkerNumberOfContainersLabelTitle());
```

Comment:
Why these are removed?

Reply:
Right now the illegal fields are highlighted with red outline so we don't need to reset label with `*` preceded.